### PR TITLE
Run status check on push with os matrix

### PR DESCRIPTION
**Pull request by matthieubosquet**
This expands the OS matrix for status check since this is a public repo and the GitHub actions' runs aren't limited anymore.

There might be failing tests in https://github.com/inrupt/artifact-generator/blob/main/src/generator/ArtifactGenerator.test.js.

### CODE REVIEW

File: .github/workflows/status-check.yml [hunk 1]
```diff
@@ -1,7 +1,4 @@
-on:
-  pull_request:
-    branches:
-    - main
+on: push
```
Comment (pmcb55) [inline .github/workflows/status-check.yml:1]: I'm not sure why this change would make things better - so asking @NSeydoux to review this part.

File: .github/workflows/status-check.yml [hunk 2]
```diff
@@ -11,7 +8,7 @@ jobs:
     strategy:
       matrix:
         # Available OS's: https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners
-        os: [ubuntu-20.04]
+        os: [ubuntu-20.04, windows-2019, macos-11]
```
Comment (pmcb55) [inline .github/workflows/status-check.yml:11]: On this, I know Jack was working to get the tests passing on Windows, not sure if he completed that ticket though.
Not sure either on MacOS - yes, we should be testing there too, but I don't have a Mac, so it would be awkward for me to verify it works there (but I guess if Matthieu is making this PR, then it works for him there...?!).

File: .github/workflows/status-check.yml [hunk 3]
```diff
@@ -11,7 +8,7 @@ jobs:
...
         node-version: [16.x, 14.x, 12.x]
     runs-on: ${{ matrix.os }}
     name: Run tests on all PRs to 'main'
```
Comment (pmcb55) [inline .github/workflows/status-check.yml:14]: This line would need updated I guess if we accept the change to the action 'on:' condition above.

File: .github/workflows/status-check.yml (hunk 1 above)
Comment (matthieubosquet) [reply .github/workflows/status-check.yml]: As discussed, it makes sense to only run on PRs to main.

**Comment by matthieubosquet** [comment]
Closing as too old (cleanup).
